File: api/server.test.js
const request = require('supertest');
const express = require('express');
const app = require('./server'); // if you export app from api.js

describe('API Endpoints', () => {
  it('GET / should return a health check message', async () => {
    const res = await request(app).get('/');
    expect(res.statusCode).toEqual(200);
    expect(res.text).toBe('API Threat Assessment Tool API is running.');
  });

  it('POST /scan should start a mock scan', async () => {
    const res = await request(app)
      .post('/scan')
      .send({ targetUrl: 'http://example.com' });

    expect(res.statusCode).toEqual(200);
    expect(res.body).toHaveProperty('status', 'Scan started');
    expect(res.body).toHaveProperty('target', 'http://example.com');
    expect(res.body).toHaveProperty('scanId');
  });

  it('GET /results/:scanId should return mock scan results', async () => {
    const res = await request(app).get('/results/12345');
    expect(res.statusCode).toEqual(200);
    expect(res.body).toHaveProperty('scanId', '12345');
    expect(Array.isArray(res.body.vulnerabilities)).toBe(true);
  });

  it('GET /status/:scanId should return mock scan status', async () => {
    const res = await request(app).get('/status/12345');
    expect(res.statusCode).toEqual(200);
    expect(res.body).toHaveProperty('scanId', '12345');
    expect(res.body).toHaveProperty('status');
    expect(res.body).toHaveProperty('progress');
  });
});

File: backend/serverTest.py
import pytest
from fastapi.testclient import TestClient
from server import app

client = TestClient(app)

def test_health_check():
    response = client.get("/")
    assert response.status_code == 200
    assert response.json()["message"] == "Scanning engine is running."

def test_start_scan():
    response = client.post("/scan", json={"target_url": "http://example.com"})
    assert response.status_code == 200
    json = response.json()
    assert json["status"] == "Scan initiated"
    assert json["target"] == "http://example.com"
    assert "scan_id" in json

def test_get_results():
    response = client.get("/results/mock-scan-id")
    assert response.status_code == 200
    json = response.json()
    assert json["scan_id"] == "mock-scan-id"
    assert "vulnerabilities" in json
    assert isinstance(json["vulnerabilities"], list)

def test_get_status():
    response = client.get("/status/mock-scan-id")
    assert response.status_code == 200
    json = response.json()
    assert json["scan_id"] == "mock-scan-id"
    assert "status" in json
    assert "progress" in json
